File: src/main/java/com/j148/backend/warning/model/Warning.java
package com.j148.backend.warning.model;
import com.j148.backend.contractor.model.Contractor;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Warning {
    @Min(value = 1, message = "Warning cannot be less than 1")
    private Long warningId;
    private LocalDateTime dateIssue;
    private WarningReason reason;
    private WarningState state;
    private Contractor contractor;

    public enum WarningReason {
        LATE, ABSENT, MISCONDUCT
    }

    public enum WarningState {
        APPEALED, ACTIVE, REMOVED, FINAL
    }
}

File: hrmsPython/models/Warning.py
from dataclasses import dataclass
from datetime import datetime
from enum import Enum
from typing import Optional

from dataclasses_json import dataclass_json

from models import Contractor


class WarningReason(Enum):
    LATE = "LATE"
    ABSENT = "ABSENT"
    MISCONDUCT = "MISCONDUCT"


class WarningState(Enum):
    APPEALED = "APPEALED"
    ACTIVE = "ACTIVE"
    REMOVED = "REMOVED"
    FINAL = "FINAL"

@dataclass_json
@dataclass
class Warnings:
    warningId: Optional[int] = None
    dateIssue: Optional[datetime] = None
    reason: Optional[WarningReason] = None
    state: Optional[WarningState] = None
    contractor: Optional[Contractor] = None
